src/correct-round.rkt

```diff
@@ -167,12 +170,21 @@
         [prec-old (in-vector (if (equal? 1 current-iter) vstart-precs vprecs))]
         [prec-new (in-vector vprecs-new)]
         [n (in-naturals)])
-    (vector-set! vrepeats n  (and (<= prec-new prec-old)
-                                  (andmap (lambda (x) (or (< x varc) (vector-ref vrepeats (- x varc))))
-                                          (rest instr)))))
-        
+    (vector-set! vrepeats n (and (<= prec-new prec-old)
+                                 (andmap (lambda (x) (or (< x varc) (vector-ref vrepeats (- x varc))))
+                                         (rest instr)))))
+  
   ; Step 4. Copying new precisions into vprecs
-  (vector-copy! vprecs 0 vprecs-new))
+  (vector-copy! vprecs 0 vprecs-new)
+  
+  ; Step 5. If precisions have not changed but the point didn't converge. A problem exists - add slack to every op
+  (when (false? (vector-member #f vrepeats))
```

Comment:
if all the precisions are not higher than from previous run

Reply:
The code is odd but whatever.